lib/puppet/util/name_validator.rb

```diff
@@ -0,0 +1,9 @@
+module Puppet::Util
```

Comment:
If you could put this under the PuppetX::Puppetlabs namespace, for instance https://github.com/puppetlabs/puppetlabs-aws/blob/master/lib/puppet_x/puppetlabs/aws.rb#L2. This is intended for extensions in custom providers like this without interfering with the main Puppet namespace.